creator-node/src/routes/files.js

```diff
@@ -7,17 +12,64 @@ const models = require('../models')
 const authMiddleware = require('../authMiddleware')
 const nodeSyncMiddleware = require('../redis').nodeSyncMiddleware
 const { logger } = require('../logging')
-let config = require('../config.js')
-let Redis = require('ioredis')
-let client = new Redis(config.get('redisPort'), config.get('redisHost'))
+const config = require('../config.js')
+const redisClient = new Redis(config.get('redisPort'), config.get('redisHost'))
+const resizeImage = require('../resizeImage')
 
 module.exports = function (app) {
-  /** Store image on disk + DB and make available via IPFS */
+  /** Store image in multiple-resolutions on disk + DB and make available via IPFS */
   app.post('/image_upload', authMiddleware, nodeSyncMiddleware, upload.single('file'), handleResponse(async (req, res) => {
-    // TODO: input validation
-    // TODO: switch to saveFileToIPFSFromFS
-    const { multihash } = await saveFileFromBuffer(req, req.file.buffer)
-    return successResponse({ 'image_file_multihash': multihash })
+    const ipfs = req.app.get('ipfsAPI')
+
+    if (!req.body.square) return errorResponseBadRequest('Must provide square boolean param in request body')
+    const squareFormat = (req.body.square === 'true')
+
+    const imageBufferOriginal = req.file.buffer
+
+    // Resize image to desired resolutions
+    const imageBuffer1000x1000 = await resizeImage(req, imageBufferOriginal, 1000, squareFormat)
+    const imageBuffer480x480 = await resizeImage(req, imageBufferOriginal, 480, squareFormat)
+    const imageBuffer150x150 = await resizeImage(req, imageBufferOriginal, 150, squareFormat)
+    
+    const imageBuffers = [imageBuffer1000x1000, imageBuffer480x480, imageBuffer150x150, imageBufferOriginal]
+
+    // Add directory with all images to IPFS
+    const resp = await ipfs.add([
+      { path: path.join(req.file.originalname, '1000x1000'), content: imageBuffer1000x1000 },
```

Comment:
(nonblocking) - if we aren't addressing with this path, but with the returned dir multihash (per offline discussion), cant this be any random guid as long as it is consistent in every object?